lib/bottle/bottle.ex

```diff
@@ -3,12 +3,30 @@ defmodule Bottle do
   A collection of helper functions used by all consuming and producer services
   """
 
-  alias Bottle.Core.V1
+  def publish({type, message}, opts \\ []) do
+    {exchange, routing_key} = exchange_and_routing_key(message)
+    encoded_message = wrap_and_encode({type, message}, opts)
 
-  def publish(target_arn, %V1.Bottle{} = bottle) do
-    bottle
-    |> V1.Bottle.encode()
+    with {:ok, channel} <- AMQP.Application.get_channel(:events) do
+      AMQP.Basic.publish(channel, exchange, routing_key, encoded_message)
+    end
+  end
+
+  defp exchange_and_routing_key(message) do
+    parts = Module.split(message.__struct__)
+    {routing_key, exchange_parts} = List.pop_at(parts, -1)
+
+    {Enum.join(exchange_parts, "."), routing_key}
+  end
+
+  defp wrap_and_encode({type, message}, opts) do
+    Bottle.Core.V1.Bottle.new(
+      request_id: Keyword.get(opts, :request_id, Bottle.RequestId.write(:http)),
+      resource: {type, message},
```

Comment:
I am probably forgetting something, but why do we have a type here? Can't that be inferred by the Bottle module being sent? `{:user_updated, %Bottle.Account.V1.UserUpdated{}}` seems redundant

Reply:
The "type" is actually the field name in the Message, the message is the value for that field. We've been passing them explicitly since we started using this stuff. We could probably deduce it from the module but there is one case that won't work and requires a breaking change: https://github.com/system76/bottle/blob/master/protos/bottle/core/v1/bottle.proto#L27

Reply:
Ahhh, I see. If we were going to do a breaking change, I guess now would be the time :shrug:

Reply:
I pushed a change, take a peek.